calcom/cal.com
# BUG: Buffer Times Not Working #3662

## What does this PR do?

This PR changes the way slots are validated. Previously it worked that way, that the booking with bufferTimes 60 minutes before and after was taking the booked slot + 2h after.

Booking: 10:00 - 11:00
beforeBufferTime: 60minutes
afterBufferTime: 60 minutes

How it was:
hours 10:00, 11:00, 12:00 were taken (are "busy")

How it is after fix:
hours 9:00, 10:00, 11:00 are taken (are "busy")

Fixes  #3662

 Loom Video: https://www.loom.com/share/58f356801e0549d4b64d2761ceab6acd


**Environment**: Staging(main branch) 

## Type of change




## How should this be tested?

Create an event with 1h buffer after and 1h buffer before.
Make a booking.
Check if the buffer times are taken into consideration when making another booking.

## Checklist



- I haven't checked if my PR needs changes to the documentation
- I haven't added tests that prove my fix is effective or that my feature works
